internal/sysfs/sock.go

```diff
@@ -0,0 +1,66 @@
+package sysfs
+
+import (
+	"net"
+	"syscall"
+
+	"github.com/tetratelabs/wazero/internal/fsapi"
+	"github.com/tetratelabs/wazero/internal/platform"
+)
+
+// SockAccept exposes syscall.Accept on POSIX systems.
+func SockAccept(f fsapi.File) (net.Conn, syscall.Errno) {
+	conn, err := f.(*listenerFile).ln.Accept()
+	if err != nil {
+		return nil, platform.UnwrapOSError(err)
+	}
+	return conn, 0
+}
+
+// SockSetNonblock exposes syscall.SetNonblock on POSIX systems.
+func SockSetNonblock(conn net.Conn) syscall.Errno {
+	tcpConn, ok := conn.(*net.TCPConn)
+	if !ok {
+		return syscall.EINVAL // FIXME: better errno?
+	}
+	syscallConn, err := tcpConn.SyscallConn()
+	if err != nil {
+		return platform.UnwrapOSError(err)
+	}
+	var actualErr error
+	// Control does not allow to return an error, but it is blocking;
+	// so it is ok to modify the external environment and setting
+	// `err` directly.
```

Comment:
I think this is actually referring to `actualErr` being set from the callback? If so it's a bit confusing to refer to `err` since there's another variable named `err` in the function.
```suggestion
	// `actualErr` directly.
```
Let me know if I'm mistaken.

Reply:
I may try to rewrite this part as it is a shell game for sure.